File: src/com/wave/hextractor/pojo/FileWithDigests.java
package com.wave.hextractor.pojo;

/**
 * File with digests.
 * @author slcantero
 */
public class FileWithDigests {
	private String name;
	private byte[] bytes;
	private String md5;
	private String sha1;
	private String crc32;
	
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public byte[] getBytes() {
		return bytes;
	}
	public void setBytes(byte[] bytes) {
		this.bytes = bytes;
	}
	public String getMd5() {
		return md5;
	}
	public void setMd5(String md5) {
		this.md5 = md5;
	}
	public String getSha1() {
		return sha1;
	}
	public void setSha1(String sha1) {
		this.sha1 = sha1;
	}
	public String getCrc32() {
		return crc32;
	}
	public void setCrc32(String crc32) {
		this.crc32 = crc32;
	}
	
}

File: src/main/kotlin/com/wave/hextractor/pojo/FileWithDigests.kt
package com.wave.hextractor.pojo

/**
 * File with digests.
 */
data class FileWithDigests(
    val name: String,
    val bytes: ByteArray,
    val md5: String,
    val sha1: String,
    val crc32: String
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as FileWithDigests

        if (name != other.name) return false
        if (md5 != other.md5) return false
        if (sha1 != other.sha1) return false
        if (crc32 != other.crc32) return false

        return true
    }

    override fun hashCode(): Int {
        var result = name.hashCode()
        result = 31 * result + md5.hashCode()
        result = 31 * result + sha1.hashCode()
        result = 31 * result + crc32.hashCode()
        return result
    }
}
